src/backend/api/Fusion.Resources.Domain/Queries/Departments/GetRelevantOrgUnits.cs

```diff
@@ -0,0 +1,382 @@
+
+                    delegatedParentManagerWithResposibility.AddRange(delegatedChildren.Children);
+                }
+
+
+
+
+                var adminClaims = user.Roles?.Where(x => x.Name.StartsWith("Fusion.Resources.Full") || x.Name.StartsWith("Fusion.Resources.Admin")).Select(x => x.Scope?.Value);
+                var readClaims = user.Roles?.Where(x => x.Name.StartsWith("Fusion.Resources.Request") || x.Name.StartsWith("Fusion.Resources.Read")).Select(x => x.Scope?.Value);
+
+
+                var retList = new List<QueryOrgUnitReason>();
+
+
+                if (isDepartmentManager) retList.Add(new QueryOrgUnitReason
+                {
+                    FullDepartment = user.FullDepartment,
+                    Reason = "Manager"
+                });
+                retList.AddRange(adminClaims.Select(dep => new QueryOrgUnitReason
+                {
+                    FullDepartment = dep ?? "*",
+                    Reason = "Write"
+                }));
+                //retList.AddRange(readClaims.Select(dep => new QueryOrgUnit
+                //{
+                //    FullDepartment = dep,
+                //    Reason = "Read"
+                //}));
+                retList.AddRange(departmentsWithResponsibility.Select(dep => new QueryOrgUnitReason
+                {
+                    FullDepartment = dep.Key,
+                    Reason = "DelegatedManager"
+                }));
+                //retList.AddRange(relevantSectors.Select(dep => new QueryOrgUnit
+                //{
+                //    FullDepartment = dep,
+                //    Reason = "RelevantSector"
+                //}));
+                //retList.AddRange(relevantDepartments.Select(dep => new QueryOrgUnit
+                //{
+                //    FullDepartment = dep.DepartmentId,
+                //    Reason = "RelevantDepartment"
+                //}));
+                if (isDepartmentManager) retList.AddRange(lineOrgDepartmentProfile?.Children.Select(dep => new QueryOrgUnitReason
+                {
+                    FullDepartment = dep.DepartmentId,
+                    Reason = "ParentManager"
+                }) ?? Array.Empty<QueryOrgUnitReason>());
+                //retList.AddRange(lineOrgDepartmentProfile?.Siblings.Select(dep => new QueryOrgUnit
+                //{
+                //    FullDepartment = dep.DepartmentId,
+                //    Reason = "RelevantSibling"
+                //}) ?? Array.Empty<QueryOrgUnit>());
+                retList.AddRange(delegatedParentManagerWithResposibility?.Select(dep => new QueryOrgUnitReason
+                {
+                    FullDepartment = dep.DepartmentId,
+                    Reason = "DelegatedParentManager"
+                }) ?? Array.Empty<QueryOrgUnitReason>());
+
+
+                var endResult = new List<QueryRelevantOrgUnit>();
+                foreach (var org in retList)
```

Comment:
you could move this section into separate function/method.
Also try to remove the clearing of reason, since you now get a fresh list of objects without reasons from the org-unit-cache